rejection-handler/src/main/java/org/motechproject/nms/rejectionhandler/service/impl/VillageHealthSubFacilityRejectionServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package org.motechproject.nms.rejectionhandler.service.impl;
+
+import org.motechproject.nms.rejectionhandler.domain.VillageHealthSubFacilityImportRejection;
+import org.motechproject.nms.rejectionhandler.repository.VillageHealthSubFacilityRejectionDataService;
+import org.motechproject.nms.rejectionhandler.service.VillageHealthSubFacilityRejectionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service("villageHealthSubFacilityRejectionService")
+public class VillageHealthSubFacilityRejectionServiceImpl implements VillageHealthSubFacilityRejectionService {
+
+    @Autowired
+    private VillageHealthSubFacilityRejectionDataService villageHealthSubFacilityRejectionDataService;
+
+    @Override
+    public void createRejectedVillageHealthSubFacility(VillageHealthSubFacilityImportRejection villageHealthSubFacilityImportRejection) {
+        villageHealthSubFacilityRejectionDataService.create(villageHealthSubFacilityImportRejection);
+    }
+
+    @Override
+    public void saveRejectedVillageHealthSubFacility(VillageHealthSubFacilityImportRejection villageHealthSubFacilityImportRejection) {
+        if(   villageHealthSubFacilityRejectionDataService.findByUniqueCode(villageHealthSubFacilityImportRejection.getVillageCode(),villageHealthSubFacilityImportRejection.getHealthSubFacilityCode()) == null){
```

Comment:
make this call outside if clause and then check if the output is null or not.. Do the same in all other places